### PR TITLE
add general build pipeline with release option

### PR DESCRIPTION
**Pull request by thebino**
Please check if the changed pipeline works.

I also kept the previous workflows for easier comparison

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+          }
+        - {
+            name: "macOS Latest Clang",
+            artifact: "macOS.tar.xz",
+            os: macos-latest,
+            cc: "clang",
+            cxx: "clang++",
+            INSTALL_LOCATION: ${{github.workspace}}/.local,
+            CACHE_LOCATION: ${{github.workspace}}/${{env.INSTALL_LOCATION}}
+          }
```
Comment (filipdutescu) [inline .github/workflows/build.yml:56]: Nothing to add here, to be fair. Looks great!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
```
[the remark this answers is not in the document]
Comment (thebino) [reply .github/workflows/build.yml:8]: I'm fine with that. Was only a generic of it.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    - name: run tests
+      run: |
+        cd build
+        ctest -C "$env:BUILD_TYPE" -VV
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ matrix.config.artifact }}
+        path: build/app*
```
[the remark this answers is not in the document]
Comment (thebino) [reply .github/workflows/build.yml:107]: It's the build directory passed through cmake by 'cmake - - build build'

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    if: contains(github.ref, 'tags/v')
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+
+    - name: Upload to Release
+      uses: meeDamian/github-release@2.0
```
[the remark this answers is not in the document]
Comment (thebino) [reply .github/workflows/build.yml:119]: The default github action can handle only a single file while this action template can handle multiple files in a path

File: .github/workflows/build.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (filipdutescu) [reply .github/workflows/build.yml:107]: And I assume `app` is the name of your project here, correct?

File: .github/workflows/build.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (filipdutescu) [reply .github/workflows/build.yml:119]: Well, for each configuration there should be 1 resulting file/archive, thus you need only 1 upload per config, right? Or am I missing something here?

[the remark this answers is not in the document]
Comment (thebino) [reply .github/workflows/build.yml:119]: If it is not a library and only a single executable, than you're right

[the remark this answers is not in the document]
Comment (filipdutescu) [reply .github/workflows/build.yml:119]: I will test ir out some more, once merged. Could you implement the change for the tag, please? Then I can merge it. Or if you lack the time let me know and I'll solve it. Cheers!

[the remark this answers is not in the document]
Comment (thebino) [reply .github/workflows/build.yml:119]: done

[the remark this answers is not in the document]
Comment (filipdutescu) [reply .github/workflows/build.yml:119]: Thank you very much for your time and contribution!

**Comment by filipdutescu** [comment]
Closes #4.

**Pull request merged**